### DISCUSSION TITLE
Can I update state in setBackgroundMessageHandler() ?

### DISCUSSION THREAD

**Discussion by GaylordP** [invertase/react-native-firebase#5089] [Q&A] [answered]
Hello and sorry my bad English I'm french.

I have a mobile application with number of notifications for user (same as Facebook, Instagram, ...) :

![Screenshot_20210327-181403](https://user-images.githubusercontent.com/10829674/112728639-8a7f6280-8f28-11eb-8c42-fc6472cfa625.jpg)

I manage to send messages with notification thanks to your library, thank you :)

I manage to update the number of notifications for user in my application when is in the background:

    messaging().setBackgroundMessageHandler(async remoteMessage => {
      /*
        addNotification = state of notifications number with state
      */
      addNotification(remoteMessage.data.data)
    })

Everything works fine but your documentation says not to update the state from `setBackgroundMessageHandler`, **so I am in error** (unless I do not understand because of my English) :

[https://rnfirebase.io/messaging/usage#background--quit-state-messages](https://rnfirebase.io/messaging/usage#background--quit-state-messages) :
> The handler must return a promise once your logic has completed to free up device resources. **It must not attempt to update any UI (e.g. via state)**

Did I do it right ?
If not, how update the number of notifications for user (the number must have changed when the user returns to the application) ?

Thanks you very much :)

**Comment by mikehardy** [comment, marked as answer]
Sorry for the delay in response - and as someone that speaks English natively but only recently learned Spanish and knows the difficulty of a non-native language, your English is fantastic. Worth saying

When we say you should not update state we mean that you should not update any React component state, we mention that because in the background and quit states where setBackgroundMessageHandler runs, there is no active Android Activity, and no components are mounted.

I think it is valid to update other types of state (for example redux or ReSub or AsyncStorage etc) and if you integrate with a notification package like react-native-push-notification or Notifee.app it may make sense to post notifications etc, but those are not React components and their related state

Hopefully that makes sense?
